Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,244 @@
 #include "simbody/internal/GeneralForceSubsystem.h"
 #include "simbody/internal/SimbodyMatterSubsystem.h"
 #include "simbody/internal/MultibodySystem.h"
+#include <iostream>
+#include <exception>
+#include <mutex>
+#include <thread>
 
 #include "ForceImpl.h"
 
+#include <memory>
+
+#define NUMNONPARALLELTHREADS 1
+#define NONPARALLELFORCESINDEX 0
 
 namespace SimTK {
 
+/** Calls calcForce() on all enabled forces. For an index of 0, this Task
+ calculates the force for all non-parallelized forces. For all other indices,
+ this Task executes a single parallelized force. */
+class CalcForcesTask : public ParallelExecutor::Task {
+public:
+
+    enum Mode {
+        All,
+        CachedAndNonCached,
+        NonCached
+    };
+
+    CalcForcesTask(ParallelExecutor& executor): m_executor(executor), m_mode(All) {}
+
+    void initialize() override {
+        // The ParallelExecutor is being executed (again). Must set these to 0
+        // so that we can properly accumulate forces.
+        // TODO Sherm uses "= Vec3(0)" etc. below; is that what I should do
+        // here? Also, I'm not resizing; could the size change
```

Comment:
The size could only change by adding more bodies to the System; that would invalidate almost everything including the State so it won't change during execution.